packages/blockchain/src/blockchain.ts

```diff
@@ -45,6 +45,37 @@ import type { BlockData } from '@ethereumjs/block'
 import type { CliqueConfig } from '@ethereumjs/common'
 import type { BigIntLike, DB, DBObject, GenesisState } from '@ethereumjs/util'
 
+/**
+ * Verkle or Merkle genesis root
+ * @param genesisState
+ * @param common
+ * @returns
+ */
+async function genGenesisStateRoot(
+  genesisState: GenesisState,
+  common: Common
+): Promise<Uint8Array> {
+  const genCommon = common.copy()
+  genCommon.setHardforkBy({
+    blockNumber: 0,
+    td: BigInt(genCommon.genesis().difficulty),
+    timestamp: genCommon.genesis().timestamp,
+  })
+  if (genCommon.gteHardfork(Hardfork.Prague)) {
```

Comment:
If I am not mistaken I think it should work also here to check for 6800 EIP activation with `common.isActivatedEIP()` (maybe double check). This is likely the cleaner and a bit more future proof solution to avoid direct Prague checks (maybe also check for other occurrences) but always do 6800 checks. Then we remain HF independent and things keep working if 6800 moves from Prague to whatever HF.

Reply:
ahh yes this makes sense as fork based checks are better in client